### ISSUE TITLE
Step method?

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/lit-llama#73]
What is the purpose of the `step` method? It doesn't seem to be used anywhere. Can it be a) removed for the sake of simplicity? Or b) used where the loss is correctly calculated using `F.cross_crossentropy` elsewhere?
 
https://github.com/Lightning-AI/lit-llama/blob/f808df1971cbac345b3d8de263a251374a53d5e9/lit_llama/model.py#L216-L220

**Comment by lantiga** [comment]
My bad, it’s a remnant from nanoGPT :-)

**Comment by carmocca** [comment]
It could be used in the training scripts if we want to

**Comment by awaelchli** [comment]
@carmocca we should be careful recommending such step methods. Currently, I would advise only going through forward() in Fabric. Due to the technical limitations of the ddp wrapper (which only wraps forward). The LM integration in Fabric has this issue https://github.com/Lightning-AI/lightning/issues/16856.

**Comment by rasbt** [comment]
To be honest, I would just remove it if it's not used. There is no downside of removing it, and it would (slightly) aid the overall goal of having a minimal, readable implementation.
